infra/feast-operator/internal/controller/services/tls.go

```diff
@@ -183,18 +190,19 @@ func (feast *FeastServices) mountTlsConfig(feastType FeastServiceType, podSpec *
 				},
 			},
 		})
-		container := &podSpec.Containers[0]
-		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
-			Name:      volName,
-			MountPath: GetTlsPath(feastType),
-			ReadOnly:  true,
-		})
+		if i, container := getContainerByType(feastType, podSpec.Containers); container != nil {
+			podSpec.Containers[i].VolumeMounts = append(podSpec.Containers[i].VolumeMounts, corev1.VolumeMount{
+				Name:      volName,
+				MountPath: GetTlsPath(feastType),
+				ReadOnly:  true,
+			})
+		}
 	}
 }
 
-func mountTlsRemoteRegistryConfig(feastType FeastServiceType, podSpec *corev1.PodSpec, tls *feastdevv1alpha1.TlsRemoteRegistryConfigs) {
+func mountTlsRemoteRegistryConfig(podSpec *corev1.PodSpec, tls *feastdevv1alpha1.TlsRemoteRegistryConfigs) {
```

Comment:
just curious, `TlsRemoteRegistryConfigs` this only for registry? not for other offline and online?

Reply:
> just curious, TlsRemoteRegistryConfigs existing only for registry?

correct... in the event a remote registry is specified by the user with TLS configured via `services.registry.remote.tls`, the online/offline containers will need the public cert (or ca) in order to communicate with said remote registry.

> not for other offline and online?

with this function the online/offline containers are configured with the remote registry's TLS volume.